### DISCUSSION TITLE
Map clocks to DBM/Zone indices

### DISCUSSION THREAD

**Discussion by senniraf** [koehlma/momba#16] [Ideas]
Hi, I'm currently evaluating the exploration engine of momba for timed automata. I got the problem that I cannot determine which index in the DBM/Zone refers to which clock. There is a function in the `clock-zones` crate that iterates through the clocks:
https://github.com/koehlma/momba/blob/4c202d4602da18971dd5a41a1794c88a293274f9/engine/crates/clock-zones/src/zones.rs#L215-L217
Could this function be made accessible from python? Or is there another way to determine which clock is mapped to which index?

**Comment by koehlma** [comment]
Exposing this function would not help as it does not provide the mapping.

The mapping itself can be reconstructed by using the `Translation` object which is in principle accessible through the explorer via `._compiled.translation`. The `Translation` contains `Declarations` which are used for the `clock_variables`:

https://github.com/koehlma/momba/blob/4c202d4602da18971dd5a41a1794c88a293274f9/momba/engine/translator.py#L859-L863

The clocks are later in the same order as in this list (starting at index 1).

I am sorry, that there currently is no more convenient way to access the mapping. What would be required is a function similar to 

https://github.com/koehlma/momba/blob/4c202d4602da18971dd5a41a1794c88a293274f9/momba/engine/explore.py#L224

which accesses the declarations in order to find the correct clock index.
